source/Cosmos.System2/Network/IPv4/IPPacket.cs

```diff
@@ -38,7 +38,8 @@ internal static void IPv4Handler(byte[] packetData)
             ARPCache.Update(ipPacket.SourceIP, ipPacket.SourceMAC);
 
             if (NetworkStack.AddressMap.ContainsKey(ipPacket.DestinationIP.Hash) == true ||
-                ipPacket.DestinationIP.Parts[3] == 255)
+                ipPacket.DestinationIP.Parts[3] == 255 // this is wrong x.x.x.255 is not always broadcast
```

Comment:
In which case?

Reply:
when subnets are not /24 in size

like with 172.16.0.0/12 172.16.0.255 is not a broadcast address

like with 192.168.0.0/16 192.168.0.255 is not a broadcast address

like with 172.16.0.0/24 172.16.0.255 is the broadcast address

like with 172.16.0.0/30 172.16.0.3 is the broadcast address

Reply:
I'm not sure why this check is even there, what does it gives us?

Reply:
its needed so we dont just drop broadcast packets